### PR TITLE
Additional humanitarian fix and test suite improvements

### PR DESCRIPTION
**Pull request by dalepotter**
Fixes [the additional issue](https://github.com/IATI/IATI-Dashboard/issues/488#issuecomment-446545092) described in [IATI/IATI-Stats#488](https://github.com/IATI/IATI-Dashboard/issues/488). 

Adding as a PR to this branch with the intention that the `andylolz/IATI-Stats:488-dashboard-issues-extras` branch becomes the full resolution for [IATI/IATI-Stats#488](https://github.com/IATI/IATI-Dashboard/issues/488) and avoids duplication of work already completed by @andylolz .  If merged, these changes will form part of https://github.com/IATI/IATI-Stats/pull/133.

Also adds a couple additional tests, and makes small refactors to the test suite, removing a duplicate test and making test docstrings consistent and explicit.

All tests pass locally.

### CODE REVIEW

**Comment by andylolz** [comment]
Thanks for this, @dalepotter! I should be able to review before the end of the week.

I did spot Wendy’s comment, but I know there’s already a test that explicitly tests the scenario Wendy describes, and asserts that it _is_ humanitarian. This one:
https://github.com/IATI/IATI-Stats/blob/6cc0f59ead53e2259dd5beb798d8b75de12d1b42/stats/tests/test_humanitarian.py#L610-L622
So either the test was always wrong, or this is a methodology change. The methodology wording on the dashboard is:

> ##### Appeal or Emergency Details
> 
> Use of humanitarian-scope element, both `@type` and `@code` must be present

File: stats/tests/test_humanitarian.py [hunk 1]
```diff
@@ -662,6 +661,27 @@ def test_humanitarian_scope_but_not_humanitarian(version):
     assert activity_stats.humanitarian()['uses_humanitarian_clusters_vocab'] == 0
 
 
+@pytest.mark.parametrize('version', ['1.01', '1.02', '1.03', '1.04', '1.05', '2.01', '2.02', '2.03', 'unknown version'])
+@pytest.mark.parametrize('hum_attrib_val_false', ['0', 'false', 'True', 'False', ''])
+def test_humanitarian_scope_but_humanitarian_is_false(version, hum_attrib_val_false):
+    """
+    Detect that even if an activity (at any version) contains a humanitarian-scope element is present, the humanitarian attribute must be present and marked as true to count.
```
Comment (andylolz) [inline stats/tests/test_humanitarian.py]: > contains a humanitarian-scope element is present

^^ small typo here – I’ll fix after merging.

**Comment by andylolz** [comment]
Okay – looking again, this all seems intuitively fine, regardless of existing tests.

The additional tests in eaf87671bb4242c7d3eeba2561c3a902b06a34c6 are excellent – thanks for those! They should really have been present in the original PR.

Merging!

File: stats/tests/test_humanitarian.py (hunk 1 above)
Comment (andylolz) [reply stats/tests/test_humanitarian.py]: Fixed in 071968ba6f8db27744a5bb014093346c8d54d4a5.

**Comment by dalepotter** [comment]
:1st_place_medal: Thanks for reviewing and merging!

**Comment by andylolz** [comment]
> Does that wording need changing, too?

^^ Looks like this has been addressed in IATI/IATI-Dashboard#530 :star2:
